zig/private/common/zig_build.bzl

```diff
@@ -219,6 +219,13 @@ def zig_build_impl(ctx, *, kind):
     global_args = ctx.actions.args()
     global_args.use_param_file("@%s")
 
+    if use_cc_common_link:
+        global_args.add_all([
+            # For now, linking with cc_common.link implies linking with libc.
+            # But this should probably be made configurable.
+            "-lc",
+        ])
+
```

Comment:
Why not just let users control this through `deps = ["@rules_zig//zig/lib:libc", ...]`? It's explicit and future proof in case of cc toolchains that don't need it.

Reply:
The issue here is that not linking with libc when `use_cc_common_link=True` gives a cryptic error for users where `main` symbol isn't found.

I see 2 solutions:
- Link with libc as you suggest
- Depend implicitely on libc when `use_cc_common_link=True` and introduce a a new build setting to disable this behavior.

WDYT ?

Reply:
Fair enough, the error is an issue.
Another thing that came to my mind: The link mode is set via build setting, i.e. not local to the target. If the libc dependency is declared at the target then it would be static across all link modes, including zig-mode. Users that don't want this would have to remember to select on the link mode on the deps, which isn't a good solution either.

Let's stick to your implementation here and introduce an off-switch separately when needed.